### ISSUE TITLE
deep_copy from subview fails on cuda

### ISSUE THREAD

**Issue by Char-Aznable** [kokkos/kokkos#4558]
This code:
```c++
#include <Kokkos_Core.hpp>
#include<iostream>
using ExecutionSpace = Kokkos::DefaultExecutionSpace;
using MemorySpace = ExecutionSpace::memory_space;
using Device = Kokkos::Device<ExecutionSpace,MemorySpace>;
using HostExecutionSpace = Kokkos::DefaultHostExecutionSpace;
using HostMemorySpace = HostExecutionSpace::memory_space;
using HostDevice = Kokkos::Device<HostExecutionSpace, HostMemorySpace>;
using Real = float;

template < class ... VS >
Real sumView2d(const Kokkos::View<Real**, VS...> m) {
  using ViewInput = std::decay_t<decltype(m)>;
  using D = typename ViewInput::device_type;
  using E = typename D::execution_space;
  Real s;
  Kokkos::parallel_reduce(
    Kokkos::MDRangePolicy<E, Kokkos::Rank<2>>(
      {0,0}, {m.extent(0), m.extent(1)}),
    KOKKOS_LAMBDA (const std::size_t& i, const std::size_t& j, Real& update) {
      update += m(i, j); 
    }, s);
  E{}.fence();
  return s;
}

int main(int argc, char* argv[]) {
  Kokkos::initialize(argc,argv); {
  const int s = 45;
  const int stepSizes[2] = {1, 10000};
  Kokkos::View<Real**, Device> v("v", s, (s + 1) * stepSizes[1]);
  Kokkos::deep_copy(ExecutionSpace{}, v, 1);
  for(int iStep = 2; iStep < v.extent_int(0); ++iStep) {
    auto vSub = Kokkos::subview(v,
      std::make_pair(0, (iStep + 1) * stepSizes[0]),
      std::make_pair(0, (iStep + 2) * stepSizes[1]));
    Kokkos::View<Real**, Device> vSubCopy("vSubCopy", vSub.extent(0), vSub.extent(1));
    Kokkos::deep_copy(ExecutionSpace{}, vSubCopy, vSub);
    ExecutionSpace{}.fence();
    auto vSubCopyHost = Kokkos::create_mirror_view_and_copy(HostExecutionSpace{}, vSubCopy);
    std::cout << sumView2d(vSub) << " " << sumView2d(vSubCopy) << " " << sumView2d(vSubCopyHost) << "\n";
  }


  } Kokkos::finalize();

}


```
raises `Cuda API error detected: cudaLaunchKernel returned (0x9)` from the deep_copy() inside the for loop and one can see it output corrupted copy judging from the output:
```
120000 120000 120000
200000 200000 200000
300000 300000 300000
420000 420000 420000
560000 560000 560000
720000 720000 720000
900000 900000 900000
1.1e+06 1.1e+06 1.1e+06
1.32e+06 1.32e+06 1.32e+06
1.56e+06 1.56e+06 1.56e+06
1.82e+06 0 0
2.1e+06 0 0
2.4e+06 0 0
2.72e+06 0 0
...
```

**Comment by crtrott** [comment]
We are looking into this: see above PR.

**Comment by PhilMiller** [comment]
Fix released in 3.6
